enhancements/aws-ipv6-dev-env.md

```diff
@@ -0,0 +1,158 @@
+components.  Many developers are already used to doing development against AWS
+clusters, so supporting AWS as an IPv6 development environment should make it
+easier for people to jump in to help debug and fix IPv6 issues.
+
+### Goals
+
+* Make creating an IPv6 development environment as easy as creating an AWS
+  cluster, by just setting one additional option to turn it on.
+* Support OKD in reaching a milestone of supporting a single stack IPv6 control
+  plane (IPv6 only).
+
+### Non-Goals
+
+* Reaching full support for IPv6 on AWS.  While the work will get us closer,
+  that is not the primary goal here.  The primary goal is to help make it
+  easier for all teams to exercise IPv6.
+
+## Proposal
+
+To enable IPv6 in your AWS environment, a developer would set the following
+environment variable before running the installer:
+
+```bash
+    export OPENSHIFT_AWS_USE_IPV6=”true”
+```
+
+This would be off by default.
+
+### AWS Network Environment
+
+AWS does not support a single-stack (IPv6 only) environment, but it does
+support a dual-stack (IPv4 and IPv6) environment, so that’s what is enabled
+here.  This is a summary of the changes to the network environment:
+
+* The VPC has IPv6 enabled and a `/56` IPv6 CIDR will be allocated by AWS.
+* Each Subnet will have an IPv6 `/64` subnet allocated to it.
+* All IPv4 specific security group rules have corresponding IPv6 rules created.
+* AWS Network Load Balancers (NLBs) do not support IPv6, so classic Elasic Load
+  Balancers (ELBs) are created, instead.
+* The IPv4 NLBs are disabled.
+* IPv6 DNS records (AAAA) are created and the IPv4 (A) records are disabled.
+* IPv6 routing is configured.  Since all instances get global IPv6 addresses,
+  NAT is not used from the instances out to the internet.
+
+### Install Configuration
+
+Here is the suggested network configuration for `install-config.yaml`:
+
+```yaml
+networking:
+  clusterNetwork:
+  - cidr: fd01::/48
+    hostPrefix: 64
+  machineCIDR: 10.0.0.0/16
+  networkType: OVNKubernetes
+  serviceNetwork:
+  - fd02::/112
+```
+
+Note that an IPv4 CIDR is still used for `machineCIDR` since AWS will provide a
+dual-stack (IPv4 and IPv6) environment.  We must specify the IPv4 CIDR and AWS
+will automatically allocate an IPv6 CIDR.
+
+### User Stories
+
+#### Story 1
+
+As an OKD developer working on an arbitrary component and with little knowledge
+about how to set up IPv6, I would like to easily create an IPv6 development
+environment to debug, test, and develop IPv6 support in my component.
+
+### Implementation Details/Notes/Constraints
+
+AWS does not support IPv6 only.  It requires IPv4 and provides the option of
+adding IPv6.  We will create a dual-stack environment, but disable IPv4
+resources over time to exercise more and more of the IPv6 control plane for
+OKD.
+
+We can never block IPv4 completely on AWS.  At a minimum, IPv4 is required for
+the AWS instances to reach the EC2 metadata API.
```

Comment:
For metadata access, the nodes just need a link-local address, and don't need a default route. (Hm... `machineCIDR: 169.254.0.0/16` ?)